packages/zoe/src/contracts/callSpread/payoffHandler.js

```diff
@@ -0,0 +1,88 @@
+// @ts-check
+import '../../../exported';
+import './types';
+
+import { E } from '@agoric/eventual-send';
+import { trade, natSafeMath } from '../../contractSupport';
+import { Position } from './position';
+import { calculateShare } from './calculateShare';
+
+const { subtract, multiply, floorDivide } = natSafeMath;
+
+const PERCENT_BASE = 100;
+const inverse = percent => subtract(PERCENT_BASE, percent);
+
+/**
+ * makePayoffHandler returns an object with methods that are useful for
+ * callSpread contracts.
+ *
+ * @type {MakePayoffHandler}
+ */
+function makePayoffHandler(zcf, seatPromiseKits, collateralSeat) {
+  const terms = zcf.getTerms();
+  const {
+    maths: { Collateral: collateralMath, Strike: strikeMath },
+    brands: { Strike: strikeBrand },
+  } = terms;
+  let seatsExited = 0;
+
+  /** @type {(positionKind: PositionKind) => Promise<Payment>} */
+  function makeOptionInvitation(dir) {
+    return zcf.makeInvitation(
+      // All we do at the time of exercise is resolve the promise.
+      seat => seatPromiseKits[dir].resolve(seat),
+      `collect ${dir} payout`,
+      {
+        position: dir,
+      },
+    );
+  }
+
+  function reallocateToSeat(position, sharePercent) {
+    seatPromiseKits[position].promise.then(seat => {
```

Comment:
Could we pass in this seat instead of getting it within the function? It seems cleaner to me to not have this function rely on `seatPromiseKits` being in scope